### PR TITLE
Clean up spellchecker findings.

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

File: documentation/source/coroutines.rst [hunk 1]
```diff
@@ -172,7 +172,7 @@ of :keyword:`yield`. Provided they are decorated with ``@cocotb.coroutine``,
 ``async def`` functions using :keyword:`await` and regular functions using
 :keyword:`yield` can be used interchangeable - the appropriate keyword to use
 is determined by which type of function it appears in, not by the
-sub-coroutinue being called.
```
Comment (eric-wieser) [inline documentation/source/coroutines.rst]: I can't help myself with this typo...

File: documentation/source/library_reference.rst [hunk 2]
```diff
@@ -2,6 +2,21 @@
...
+    BinaryRepresentation
+    binaryRepresentation
+    sim
+    SimHandle
+    Todo
+    un
+    arg
+    idx
+    Args
+    Enum
```
Comment (eric-wieser) [inline documentation/source/library_reference.rst]: Are all of these needed? Looks like some of them should be marked up instead. Or are they section titles?

File: documentation/source/coroutines.rst (hunk 1 above)
Comment (cmarqu) [reply documentation/source/coroutines.rst]: Seems you have all the best words! ;)

File: documentation/source/library_reference.rst (hunk 2 above)
Comment (cmarqu) [reply documentation/source/library_reference.rst]: I now removed them and mysteriously spell checking still is fine, while it wasn't before... no idea why.

**Pull request merged**